src/compiler/checker.ts

```diff
@@ -3160,6 +3160,9 @@ namespace ts {
                 if (declaration.kind === SyntaxKind.ExportAssignment) {
                     return links.type = checkExpression((<ExportAssignment>declaration).expression);
                 }
+                if (declaration.kind === SyntaxKind.JSDocPropertyTag && (<JSDocPropertyTag>declaration).typeExpression) {

```

Comment:
That method is originally for declarations written in the code whose type annotation in another seperated jsdoc comment, so other methods who call it all assumes `VariableLikeDeclaration` type of the declaration. It seems to be a lot of change to make all of them accept `VariableLikeDeclaration | JSDocPropertyTag`. What would be the benefit of doing that?